Basic-Car-Maintenance/Shared/Odometer/Views/EditOdometerReadingDetailView.swift

```diff
@@ -0,0 +1,120 @@
+                Section {
+                    HStack {
+                        TextField("Distance", value: $distance, format: .number)
+                        Button("Unit") {
+                            switchUnitModalIsPresented = true
+                        }
+                    }
+                    .sheet(isPresented: $switchUnitModalIsPresented) {
+                        Toggle(isOn: $unitsAreMetric) {
+                            Text(unitsAreMetric ? "Kilometers" : "Miles")
+                        }
+                        .presentationDetents([.height(100)])
+                    }
+                } header: {
+                    Text("Distance in \(unitsAreMetric ? "Kilometers" : "Miles")")
+                }
+                
+                Section {
+                    Picker(selection: $selectedVehicle) {
+                        ForEach(viewModel.vehicles) { vehicle in
+                            Text(vehicle.name).tag(vehicle as Vehicle)
+                        }
+                    } label: {
+                        Text("Select a vehicle",
+                             comment: "Odometer Reading vehicle picker label")
+                    }
+                    .pickerStyle(.menu)
+                } header: {
+                    Text("Vehicle",
+                         comment: "Odometer reading vehicle picker header")
+                }
+                
+                DatePicker(selection: $date, displayedComponents: .date) {
+                    Text("Date")
+                }
+            }
+            .onAppear {
+                guard let selectedReading = selectedReading else { return }
+                setOdometerReadingValues(reading: selectedReading)
+            }
+            .navigationTitle(Text("Update Odometer Reading"))
+            .toolbar {
+                ToolbarItem(placement: .topBarLeading) {
+                    Button {
+                        dismiss()
+                    } label: {
+                        Text("Cancel")
+                    }
+                }
+                
+                ToolbarItem(placement: .topBarTrailing) {
+                    Button {
+                        if let selectedVehicle, let selectedReading {
+                            var reading = OdometerReading(date: date,
+                                                          distance: distance,
+                                                          unitsAreMetric: unitsAreMetric,
+                                                          vehicle: selectedVehicle)
+                            reading.id = selectedReading.id
+                            Task {
+                                await viewModel.updateReading(reading)
+                                dismiss()
+                            }
+                        }
+                    } label: {
+                        Text("Update")
+                    }
+                    .disabled(distance == 0)
```

Comment:
You could change this to `distance < 0`, to make sure someone doesn't try to enter a negative number

Reply:
fixed in https://github.com/mikaelacaron/Basic-Car-Maintenance/pull/175/commits/039d3c4000c96056b12ac420ebcba489162643bc